### PR TITLE
Signup refactor

### PR DESCRIPTION
**Pull request by damfinkel**
## Summary

- Added react-hook-forms to React SignUp screen
- Added basic validations
- Added tests
- Fixed styling because fields were misaligned
- Removed login/logout actionCreators both of which weren't doing anything
- Small improvements to login and tests

## Screenshots

pivot

### CODE REVIEW

File: cookbook-react/src/recipes/screens/login/index.test.tsx [hunk 1]
```diff
@@ -104,9 +102,10 @@ describe('#Login', () => {
         await fireEvent.submit(form);
       });
 
-      expect(mockLogin).toHaveBeenCalled();
       expect(mockSetStateUser).toHaveBeenCalled();
+      expect(mockSetStateUser).toHaveBeenCalledWith({ sessionToken: 'token', id: 1234 });
```
Comment (mcallegari10) [inline cookbook-react/src/recipes/screens/login/index.test.tsx:106]: Is it really necessary to test if it was called AND it was called with params? I think the `toHaveBeenCalledWith` is enough 🤔

File: cookbook-react/src/recipes/screens/login/index.test.tsx [hunk 2]
```diff
@@ -104,9 +102,10 @@ describe('#Login', () => {
...
       expect(mockSetPersistantUser).toHaveBeenCalled();
+      expect(mockSetPersistantUser).toHaveBeenCalledWith({ sessionToken: 'token', id: 1234 });
```
Comment (mcallegari10) [inline cookbook-react/src/recipes/screens/login/index.test.tsx:108]: Same for this

File: cookbook-react/src/recipes/screens/login/index.test.tsx [hunk 3]
```diff
@@ -94,19 +92,18 @@ describe('#Login', () => {
...
-      const form = await getByRole('form', { name: 'login-form' });
+      const email = getByLabelText('Login:email');
+      const password = getByLabelText('Login:password');
+      const form = getByRole('form', { name: 'login-form' });
 
       await act(async () => {
         await fireEvent.change(email, { target: { value: 'someone@wolox.com' } });
         await fireEvent.change(password, { target: { value: 'myPassword' } });
         await fireEvent.submit(form);
       });
```
Comment (pabloferro) [inline cookbook-react/src/recipes/screens/login/index.test.tsx:103]: Is act necessary here?

File: cookbook-react/src/recipes/screens/registration/index.tsx [hunk 4]
```diff
@@ -4,76 +4,116 @@ import i18next from 'i18next';
...
+import { useLazyRequest } from '~hooks/useRequest';
+import { signup, setCurrentUser, RegistrationUser } from '~services/AuthServices';
+import { actionCreators } from '~contexts/UserContext/reducer';
+import { User } from '~app/contexts/UserContext/reducer';
+import { useHistory } from 'react-router';
 
-const FIELDS = stringArrayToObject(['name', 'surname', 'email', 'password', 'confirmPassword']);
+const FIELDS = stringArrayToObject(['firstName', 'lastName', 'email', 'password', 'confirmPassword']);
+
+const mandatoryValidation = { required: i18next.t('Registration:requiredError') as string };
```
Comment (pabloferro) [inline cookbook-react/src/recipes/screens/registration/index.tsx:19]: Why not `requiredValidation`?

File: cookbook-react/src/recipes/screens/registration/index.tsx [hunk 5]
```diff
@@ -4,76 +4,116 @@ import i18next from 'i18next';
...
+  firstName: mandatoryValidation,
+  lastName: mandatoryValidation,
+  email: { 
+    pattern: { 
+      value: /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+      message: i18next.t('Registration:emailFormatError') as string
+    },
+    ...mandatoryValidation
+  },
+  password: { ...mandatoryValidation, minLength: { value: 8, message: i18next.t('Registration:passwordLengthError') as string } },
```
Comment (pabloferro) [inline cookbook-react/src/recipes/screens/registration/index.tsx:31]: Extract 8 to a const and pass it to the translation.

File: cookbook-react/src/recipes/screens/registration/index.tsx [hunk 6]
```diff
@@ -4,76 +4,116 @@ import i18next from 'i18next';
...
         <FormInput
-          className="full-width m-left-1"
+          className={`${styles.inputContainer} item-1`}
           label={i18next.t('Registration:confirmPassword')}
           name={FIELDS.confirmPassword}
           inputType="password"
           inputClassName={styles.input}
           placeholder={i18next.t('Registration:confirmPasswordPlaceholder') as string}
-          onChange={handleInputChange}
+          inputRef={register({ ...VALIDATIONS.confirmPassword, validate: validateConfirmPassword })}
```
Comment (pabloferro) [inline cookbook-react/src/recipes/screens/registration/index.tsx:115]: Why isn't `validateConfirmPassword', part of `VALIDATIONS.confirmPassword`?

**Review by pabloferro — CHANGES_REQUESTED** [review]
See if you can remove the `act`s

File: cookbook-react/src/recipes/screens/registration/index.tsx (hunk 4 above)
Comment (damfinkel) [reply cookbook-react/src/recipes/screens/registration/index.tsx:19]: mandatory and required are synonyms

File: cookbook-react/src/recipes/screens/registration/index.tsx (hunk 6 above)
Comment (damfinkel) [reply cookbook-react/src/recipes/screens/registration/index.tsx:115]: because it uses `watch`, which is returned by the useForm hook from RHF.

**Review by pabloferro — DISMISSED** [review]

**Pull request merged**